io-engine/src/rebuild/rebuild_error.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{bdev_api::BdevError, core::CoreError};
+use snafu::Snafu;
+use spdk_rs::{BdevDescError, DmaError};
+
+#[derive(Debug, Snafu, Clone)]
+#[snafu(visibility(pub(crate)), context(suffix(false)))]
```

Comment:
what does `context(suffix(false))` do?

Reply:
New version of Snafu adds a suffix "Error" for context structs it generates. E.g., for Error::NexusNotFound enum, Snafu generates a struct named "NexusNotFoundError" to be used with .context() method.
This breaks compatibility with existing error enums. The easiest way is to revert to old behaviour by context(suffix(false)).
But, there's a caveat: although Snafu won't add "Error" suffix with this option, it still _removes_ Error if it is already there. E.g. for MyError::NameExistsError enum Snafu generates "NameExists" struct.
Fortunately, we only had a few enums ending in "Error" in data plane, so it was not hard to rename them.

The reason why I updated to newer Snafu is that it allows to put generated struct in a namespace, preventing module's namespace pollution, and making exporting error definitions easier.

Reply:
Thanks for explaining :+1: we might also upgrade snafu on control-plane at some point